code/datums/mind.dm

```diff
@@ -980,32 +977,21 @@
 		switch(href_list["vampire"])
 			if("clear")
 				if(src in SSticker.mode.vampires)
-					SSticker.mode.vampires -= src
-					special_role = null
-					if(vampire)
-						vampire.remove_vampire_powers()
-						qdel(vampire)
-						vampire = null
-					SSticker.mode.update_vampire_icons_removed(src)
+					remove_antag_datum(/datum/antagonist/vampire)
 					to_chat(current, "<FONT color='red' size = 3><B>You grow weak and lose your powers! You are no longer a vampire and are stuck in your current form!</B></FONT>")
 					log_admin("[key_name(usr)] has de-vampired [key_name(current)]")
 					message_admins("[key_name_admin(usr)] has de-vampired [key_name_admin(current)]")
 			if("vampire")
 				if(!(src in SSticker.mode.vampires))
```

Comment:
We should go ahead and start moving away from the `X in SSticker.mode.vampires` (or thralls) checks and start only checking if they have the antag datum. We want to be the least reliant on the gamemode as possible, until we can completely disconnect them.